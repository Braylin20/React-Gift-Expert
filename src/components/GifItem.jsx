

export const GifItem = ({ title, url, onDelete }) => {


    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
            <button onClick={onDelete}>X</button>
        </div>
    )
}
