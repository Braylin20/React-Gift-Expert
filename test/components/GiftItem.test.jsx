/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"



describe('Pruebas en <GifItem', () => {

    const title = 'Dragon ball'
    const url = 'https://dragonball'

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('should show the image and url correct', () => {
        render(<GifItem title={title} url={url} />)
        const { src, alt } = screen.getByRole('img');
        screen.debug()
        // expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should show the title of component', () => {

    })
})