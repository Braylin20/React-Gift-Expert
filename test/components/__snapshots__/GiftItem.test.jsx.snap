// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://dragonball"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
